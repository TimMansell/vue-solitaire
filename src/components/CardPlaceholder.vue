<template>
  <SvgIcon :class="classes" data-test="card-placeholder" name="Card_back_17" />
</template>

<script>
import SvgIcon from '@/components/SvgIcon.vue';

export default {
  name: 'CardPlaceholder',
  components: {
    SvgIcon,
  },
  props: {
    seeThrough: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      const { seeThrough } = this;

      return {
        'card-placeholder': seeThrough,
      };
    },
  },
};
</script>

<style scoped lang="scss">
.card-placeholder {
  opacity: 0.15;
}
</style>
