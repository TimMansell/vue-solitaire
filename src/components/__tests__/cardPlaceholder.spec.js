import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import CardPlaceholder from '@/components/CardPlaceholder.vue';

describe('CardPlaceholder.vue', () => {
  it('renders the component without crashing', () => {
    const wrapper = shallowMount(CardPlaceholder);

    expect(wrapper.isVisible()).toBe(true);
  });

  it('should have correct props', () => {
    const wrapper = shallowMount(CardPlaceholder, {
      props: {
        seeThrough: true,
      },
    });

    expect(wrapper.props().seeThrough).toBe(true);
  });

  it('should render a see-through card', () => {
    const wrapper = shallowMount(CardPlaceholder, {
      props: {
        seeThrough: true,
      },
    });

    expect(wrapper.classes()).toContain('card-placeholder');
  });

  it('should show 1 placeholder card', () => {
    const wrapper = shallowMount(CardPlaceholder, {
      props: {
        seeThrough: true,
      },
    });

    expect(wrapper.findAll('[data-test="placeholder-card"]')).toHaveLength(1);
  });

  it('should show multiple column placeholder cards', () => {
    const wrapper = shallowMount(CardPlaceholder, {
      props: {
        cards: 5,
      },
    });

    expect(wrapper.findAll('[data-test="placeholder-card"]')).toHaveLength(5);
  });
});
