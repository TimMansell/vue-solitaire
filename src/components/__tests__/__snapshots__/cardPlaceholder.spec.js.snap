// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPlaceholder.vue matches snapshot 1`] = `<svgicon-stub name="Card_back_17" data-test="card-placeholder" class=""></svgicon-stub>`;
